### DISCUSSION TITLE
Composer v1.x for the latest version of Docker-Magento

### DISCUSSION THREAD

**Discussion by st3phani** [markshust/docker-magento#510] [Q&A] [answered]
How can you replace the latest version of `docker-magento` in **Composer v2.x** in **v1.x**?
I modified from:

```
phpfpm:
  image: markoshust/magento-php:7.4-fpm-6
```
into
```
phpfpm:
  image: markoshust/magento-php:7.4-fpm-5
```
but does not work NGNIX.

> I want to change the version, because there are some Third-Party composer packages that do not work with Composer v2.x

**Comment by markshust** [comment, marked as answer]
The best way to downgrade Composer is to keep the newer image, but run:

```
bin/root composer self-update --1
```

This will downgrade the project back to Composer 1, and it should persist as long as the volumes aren't removed.

**Comment by riccardofinkers** [comment]
Hi @markshust , 

This is just what I needed ! I have some old projects running on 2.3.x or on 2.4.1 and there I need to downgrade composer to V1. 

Now I have a project where I used the latest version of `markoshust/magento-php:7.3-fpm` and then ran `bin/root composer self-update --1` 
but unfortunately it gave me the following error so I now work on `markoshust/magento-php:7.3-fpm-12` but that version is not so fast on my M1 so I hope I can downgrade composer and still be on the newest version. 

The error after running `bin/root composer self-update --1`. Btw I also get the same error when running just `bin/root composer` after downgrading to 7.3-fpm-12 this message is gone
````
Fatal error: Class Magento\InventoryComposerInstaller\Plugin contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (Composer\Plugin\PluginInterface::deactivate, Composer\Plugin\PluginInterface::uninstall) in /var/www/html/vendor/magento/inventory-composer-installer/src/Plugin.php on line 23

````

**Reply by markshust** [reply]
Hi @riccardofinkers, have you attempted to remove or disable the `inventory-composer-installer` module after upgrading or downgrading? This appears to be caused by code rather than anything specific to this project or Docker image. I've noticed some funky things going on sometimes with Magento modules and upgrades/downgrades/software versions.
